engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -661,15 +658,33 @@ private String paramsTypesToString(Class<?>[] paramTypes) {
         return buf.toString();
     }
 
+    private static class CandidateExecutable<EXECUTABLE_TYPE extends Executable> {
+        private final EXECUTABLE_TYPE executable;
+        private final int cost;
+
+        private CandidateExecutable(final EXECUTABLE_TYPE executable, final int cost) {
+            this.executable = executable;
+            this.cost = cost;
+        }
+
+        private static final int NO_CONVERSION_POSSIBLE = -1;
+        private static final int ALREADY_EQUAL = 0;
+        private static final int BOX_CONVERSION = 1;
+        private static final int WIDEN_CONVERSION = 2;
+        private static final int UNBOX_CONVERSION = 5; // may NPE, so is more expensive
+        private static final int VECTOR_CONVERSION = 10;
```

Comment:
Vector conversion is arbitrarily expensive. Maybe make it 100?
I think unboxing goes before boxing, which goes before widening, to match the JVM. The NPE is specific to `Boolean`, we think. Maybe update comment accordingly.
```
        private static final int UNBOX_CONVERSION = 5; // may NPE, so is more expensive
        private static final int VECTOR_CONVERSION = 10;
```